src/bin/pg_autoctl/cli_drop_node.c

```diff
@@ -851,3 +868,49 @@ stop_postgres_and_remove_pgdata_and_config(ConfigFilePaths *pathnames,
 		exit(EXIT_CODE_BAD_CONFIG);
 	}
 }
+
+
+/*
+ * wait_until_node_is_dropped waits until the node doesn't exist anymore on the
+ * monitor, meaning it's been fully dropped now.
+ */
+static void
+wait_until_node_is_dropped(KeeperConfig *config, int64_t nodeId, int groupId)
+{
+	bool dropped = false;
+	Monitor monitor = { 0 };
+
+	(void) cli_monitor_init_from_option_or_config(&monitor, config);
+
+
+	while (!dropped)
+	{
```

Comment:
Should have some check that calls `exit` after some timeout (probably `PG_AUTOCTL_LISTEN_NOTIFICATIONS_TIMEOUT`). Otherwise we can get stuck forever here.

Reply:
Good point also, added too.